platform/viewer/src/connectedComponents/ViewerRetrieveStudyData.js

```diff
@@ -55,17 +56,28 @@ class ViewerRetrieveStudyData extends Component {
           );
         }
         studyMetadata.setDisplaySets(study.displaySets);
+
         // Updates WADO-RS metaDataManager
         updateMetaDataManager(study);
         studyMetadataManager.add(studyMetadata);
+
         // Attempt to load remaning series if any
-        this._attemptToLoadRemainingSeries(studyMetadata);
+        this._loadRemainingSeries(studyMetadata).then(
+          this._studyDidLoad.bind(this)
+        );
+
         return study;
       });
       this.setState({ studies });
     }
   }
 
+  _studyDidLoad() {
+    this.setState({
+      seriesLoaded: true,
```

Comment:
This should be `studyLoaded`? Its for multiple series (and the same wherever else its used).

Reply:
Yes, it can be studyLoaded. 
I've added seriesLoaded just because I was waiting all series promises.

Reply:
I'm fine with `studyLoaded` or `allSeriesLoaded`, it was just a little confusing  as series it was a bit ambiguous as to whether `seriesLoaded` was singular or plural.

Since you already changed it, if you are happy with `studyLoaded` we can go with that.